app/controllers/tasks_controller.rb

```diff
@@ -60,4 +62,24 @@ def task_params
   def dashboard_path
     current_user.mentee? ? dashboard_mentee_profiles_path(week: @task.week) : dashboard_mentor_profiles_path(week: @task.week)
   end
+
+  def award_novice_badge
+      if @task.project.tasks.finished(current_user).count >= 1
+           assign_novice_badge
+      else unassign_novice_badge
```

Comment:
why?

Reply:
I guess why is for why we are  unassigning the badge ? 
Well this is because, in tasks, someone can untick the task, then task goes from fnshed to unfinished again, if its the first task, then badge need to be removed.

Reply:
@DMips Also, important to consider; are we planning to handle unassignments as well ?

Reply:
@anjali-sharma we can unassign badge for tasks but overall changes should be irreversible so no unassignment should be needed . We can disable project description edition once corresponding badge is awarded by organizer to make sure assignment is valid

Reply:
Sounds good. Let's do that. Disabling description edit should solve the problem.